src/components/actionSheet/actionSheet.js

```diff
@@ -0,0 +1,372 @@
+		dialogOptions.autoFocus = true;
+	} else {
+
+		dialogOptions.modal = false;
+		dialogOptions.entryAnimation = options.entryAnimation;
+		dialogOptions.exitAnimation = options.exitAnimation;
+		dialogOptions.entryAnimationDuration = options.entryAnimationDuration || 140;
+		dialogOptions.exitAnimationDuration = options.exitAnimationDuration || 100;
+		dialogOptions.autoFocus = false;
+	}
+
+	var dlg = dialogHelper.createDialog(dialogOptions);
+
+	if (isFullscreen) {
+		dlg.classList.add('actionsheet-fullscreen');
+	} else {
+		dlg.classList.add('actionsheet-not-fullscreen');
+	}
+
+	dlg.classList.add('actionSheet');
+
+	if (options.dialogClass) {
+		dlg.classList.add(options.dialogClass);
+	}
+
+	var html = '';
+
+	var scrollClassName = layoutManager.tv ? 'scrollY smoothScrollY hiddenScrollY' : 'scrollY';
+	var style = '';
+
+	// Admittedly a hack but right now the scrollbar is being factored into the width which is causing truncation
+	if (options.items.length > 20) {
+		var minWidth = dom.getWindowSize().innerWidth >= 300 ? 240 : 200;
+		style += "min-width:" + minWidth + "px;";
+	}
+
+	var i;
+	var length;
+	var option;
+	var renderIcon = false;
+	var icons = [];
+	var itemIcon;
+	for (i = 0, length = options.items.length; i < length; i++) {
+
+		option = options.items[i];
+
+		itemIcon = option.icon || (option.selected ? 'check' : null);
+
+		if (itemIcon) {
+			renderIcon = true;
+		}
+		icons.push(itemIcon || '');
+	}
+
+	if (layoutManager.tv) {
+		html += '<button is="paper-icon-button-light" class="btnCloseActionSheet hide-mouse-idle-tv" tabindex="-1"><i class="material-icons arrow_back"></i></button>';
+	}
+
+	// If any items have an icon, give them all an icon just to make sure they're all lined up evenly
+	var center = options.title && (!renderIcon /*|| itemsWithIcons.length != options.items.length*/);
+
+	if (center || layoutManager.tv) {
+		html += '<div class="actionSheetContent actionSheetContent-centered">';
+	} else {
+		html += '<div class="actionSheetContent">';
+	}
+
+	if (options.title) {
+
+		html += '<h1 class="actionSheetTitle">';
+		html += options.title;
+		html += '</h1>';
+	}
+	if (options.text) {
+		html += '<p class="actionSheetText">';
+		html += options.text;
+		html += '</p>';
+	}
+
+	var scrollerClassName = 'actionSheetScroller';
+	if (layoutManager.tv) {
+		scrollerClassName += ' actionSheetScroller-tv focuscontainer-x focuscontainer-y';
+	}
+	html += '<div class="' + scrollerClassName + ' ' + scrollClassName + '" style="' + style + '">';
+
+	var menuItemClass = 'listItem listItem-button actionSheetMenuItem';
+
+	if (options.border || options.shaded) {
+		menuItemClass += ' listItem-border';
+	}
+
+	if (options.menuItemClass) {
+		menuItemClass += ' ' + options.menuItemClass;
+	}
+
+	if (layoutManager.tv) {
+		menuItemClass += ' listItem-focusscale';
+	}
+
+	if (layoutManager.mobile) {
+		menuItemClass += ' actionsheet-xlargeFont';
+	}
+
+	for (i = 0, length = options.items.length; i < length; i++) {
```

Comment:
Same for this :)

Reply:
can you check again here multiple variables are there where `i` is referenced here its looped over items and inside the loop it is used for icons to to get the respective icon.